### ISSUE TITLE
Option of `distclean` command to remove obsolete/unnecessary distfiles

### ISSUE THREAD

**Issue by yasuhirokimura** [freebsd/poudriere#838]
For those who build custom packages with poudriere, it is typical usage to create file that includes list of ports he want to build and use it with -f options of `bulk` command.

```
# cat ports-list.txt
cat1/aaa
cat2/bbb
cat3/ccc
...

# poudriere bulk -j jailname -f ports-list.txt
```

When a port is updated, sometimes dependency changes. For example dependency of `cat1/aaa` may change from `cat10/xxx` to `cat20/yyy`. In this case `xxx` package in package repository gets obsolete and isn't used any more. If `cat10/xxx` is updated `xxx` package in package repository is automatically removed when `bulk` command is executed next time. But it is possible to explicitly remove it by using `pkgclean` command.

```
# poudriere pkgclean -j jailname -f ports-list.txt
```

With above command obsolete packages are removed from the repository.

Now let me get into the main subject.

When dependency changed and some packages get obsolete, distfiles for them also get obsolete. But different from packages distfiles are never automatically removed with `bulk` command. There is `distclean` command but unfortunately it doesn't seem to be intended to be used for this purpose. If I execute `poudriere distclean -f ports-list.txt`, then distfiles necessary to build ports listed in `ports-list.txt` and it dependency are removed. To be honest, I can't imagine why the behavior of the `distclean` command was designed this way. IMHO `distclean` command should behave so it removes distfiles unnecessary to build ports listed in the file specified with '-f' options. But changing the behavior may result in unexpected problem. So please consider providing option of `distclean` command that changes the behavior in this way.

Best Regards.

**Comment by bdrewery** [comment]
> If I execute poudriere distclean -f ports-list.txt, then distfiles necessary to build ports listed in ports-list.txt and it dependency are removed

This doesn't sound right. If I create a file `UNEXPECTED` in my distdir and run `poudriere distclean -n -f mylist` I see `UNEXPECTED` in the list of files to delete.

> IMHO distclean command should behave so it removes distfiles unnecessary to build ports listed in the file specified with '-f' options.

I might not be understanding you because this is how it works.

The way it is intended to work is to keep all distfiles for the listed ports and its dependencies. Everything else is removed. When I use `touch empty; poudriere distclean -n -f empty` it wants to remove all of my distfiles.

**Comment by bdrewery** [comment]
I'm fairly certain commit fb8d95e3b01c774e4e0fdc205b0fad57851b3400 fixes the issues seen here. `distclean` was deleting incorrectly deleting random valid files.

**Comment by bdrewery** [comment]
I do see an issue with dependencies that will be addressed in #919.
